File: src/main/scala/com/vc/easy/L563.scala
package com.vc.easy

class TreeNode563(var _value: Int) {
  var value: Int = _value
  var left: TreeNode563 = null
  var right: TreeNode563 = null
}

object L563 {
  def findTilt(root: TreeNode563): Int = {
    var sumTilt = 0
    def solve(root: TreeNode563): Int = {
      if(root != null) {
        val l = solve(root.left)
        val r =  solve(root.right)
        sumTilt += Math.abs(l - r)
        return l + r + root.value
      }
      0
    }
    solve(root)
    sumTilt
  }
}

File: src/main/java/com/vc/easy/L563.java
package com.vc.easy;

import com.vc.hard.TreeNode;

class L563 {
    int res = 0;
    public int findTilt(TreeNode root) {
        helper(root);
        return res;
    }

    private int helper(TreeNode root) {
        if(root == null) return 0;

        int left = helper(root.left);
        int right = helper(root.right);

        res += Math.abs(left - right);

        return left + right + root.val;
    }
}
